lib/noether/list.ex

```diff
@@ -45,4 +46,41 @@ defmodule Noether.List do
         [a] ++ until(p, f, f.(a))
     end
   end
+
+  @doc """
+  Given a list, it returns `{:ok, list}` if every element of the list is different from nil or error. Otherwise `{:error, :nil_found}` is returned.
```

Comment:
```suggestion
  Given a list, it returns `{:ok, list}` if every element of the list is different from nil or `{:error, _}`. Otherwise `{:error, :nil_found}` is returned.
```